sources/packages/backend/apps/api/src/services/education-program-offering/custom-validators/has-valid-funded-weeks-for-offering-credentials.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  ValidatorConstraintInterface,
+  ValidatorConstraint,
+  registerDecorator,
+  ValidationOptions,
+  ValidationArguments,
+} from "class-validator";
+import {
+  OfferingValidationModel,
+  OfferingYesNoOptions,
+  StudyBreak,
+} from "../education-program-offering-validation.models";
+import { OfferingCalculationValidationBaseConstraint } from "./offering-calculation-validation-base-constraint";
+
+/**
+ * For an offering that contains study breaks, execute the calculation
+ * of the funded study period to validate if it matches the maximum allowed
+ * study period amount of funded weeks for the selected offering credential.
+ */
+@ValidatorConstraint()
+class HasValidFundedWeeksForOfferingCredentialsConstraint
```

Comment:
IMO `HasFundedWeeksWithinMaximumLimit` could be more meaningful as the constraint itself is not based on a credential.